src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -210,10 +216,12 @@ public X509Certificate getCACertificate() throws KeyStoreException {
       *
       * @param privateKey  leaf CA private key
       * @param caCertificateChain a CA chain
+      * @param providerType  provider type DEFAULT or HSM, used to map to the correct JCA provider
       *
       * @throws KeyStoreException  if privateKey is not instance of PrivateKey or no ca chain provided
       */
-     public synchronized void setCaKeyAndCertificateChain(Key privateKey, X509Certificate... caCertificateChain)
+     public synchronized void setCaKeyAndCertificateChain(
+             CertificateHelper.ProviderType providerType, Key privateKey, X509Certificate... caCertificateChain)
```

Comment:
I don't think we have the right interface here in the CertificateStore. Ideally we'd just encapsulate everything. But, I'm ok with this for now. I hate how we're leaking details like the underlying crypto provider

Reply:
There was not a really good way of checking if we should use the HSM provider since that is basically dictated by configuration and I didn't want to add more stuff. I am sure we could work something out to improve it.